src/submission/submission-set-helper.service.ts

```diff
@@ -0,0 +1,156 @@
+import { Injectable } from '@nestjs/common';
+import { EntityManager } from 'typeorm';
+import { SubmissionSet } from '../entities/submission-set.entity';
+import { SubmissionQueue } from '../entities/submission-queue.entity';
+import { MonitorPlan } from '../entities/monitor-plan.entity';
+import { QaCertEvent } from '../entities/qa-cert-event.entity';
+import { QaTee } from '../entities/qa-tee.entity';
+import { EmissionEvaluation } from '../entities/emission-evaluation.entity';
+import { MatsBulkFile } from '../entities/mats-bulk-file.entity';
+import { Logger } from '@us-epa-camd/easey-common/logger';
+import { QaSuppData } from '../entities/qa-supp.entity';
+import { Plant } from '../entities/plant.entity';
+
+@Injectable()
+export class SubmissionSetHelperService {
+  constructor(
+    private readonly entityManager: EntityManager,
+    private readonly logger: Logger,
+  ) {}
+
+  async updateSubmissionSetStatus(
+    submissionSet: SubmissionSet,
+    statusCode: string,
+    note: string = '',
+  ): Promise<void> {
+    submissionSet.statusCode = statusCode;
+    if (note) {
+      submissionSet.note = note;
+      submissionSet.noteTime = new Date();
+    }
+
+    if (statusCode === 'WIP') {
+      submissionSet.startedTime = new Date();
+    }
+
+    if (statusCode === 'COMPLETE') {
+      submissionSet.completedTime = new Date();
+    }
+
+    await this.entityManager.save(submissionSet);
+  }
+
+  async setRecordStatusCode(
+    set: SubmissionSet,
+    records: SubmissionQueue[],
+    statusCode: string,
+    note: string,
+    originRecordCode: string,
+  ): Promise<void> {
+    for (const record of records) {
+      record.statusCode = statusCode;
+
+      if (note) {
```

Comment:
Not sure but it seems like this should check for statusCode === 'ERROR' in addition to note is not null.  Right now a caller could bas "COMPLETE' or 'WIP' with an actual Note, and startedTime or completedTime and both note and NoteTime could be set at the same time.  I believe the call should only do the setting of values for one status in a single call.

**I am definitely looking for pushback on this comment!**

Reply:
This method (and class) is shared across submission processing (setting it to WIP when submission starts, setting status to COMPLETE when submission ends) and error handling.  For statusCode === 'ERROR', note is passed (whereas for WIP and COMPLETE, a note is not passed as there is no error - the note field is used for errors only).   So, this is operating as intended.